### ISSUE TITLE
Tabs are breaking in combination with navigation bar

### ISSUE THREAD

**Issue by aurosevic** [flet-dev/flet#2109]
**Description**
After changing the view using the navigation bar to another 'NavigationDestination' and then returning to the one with tabs, the tabs completely break and become unresponsive. This issue occurs consistently every time I follow this sequence.

https://github.com/flet-dev/flet/assets/19304661/570a9520-8a5b-4823-b5e5-247a6de7a73b

**Code example to reproduce the issue:**

```python
flet run app.py --verbose
```

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important (e.g. issue happens only occasionally):**

**Flet version (`pip show flet`)**:

```
Name: flet
Version: 0.12.2
Summary: Flet for Python - easily build interactive multi-platform apps in Python
Home-page: 
Author: Appveyor Systems Inc.
Author-email: hello@flet.dev
License: Apache-2.0
Location: /home/andrija/PycharmProjects/finance-buddy/venv/lib/python3.10/site-packages
Requires: cookiecutter, flet-runtime, packaging, qrcode, watchdog, websocket-client, websockets
Required-by: 
```

**Operating system**:

KDE Neon 5.27

**Additional environment details:**

```
(venv) andrija@multiverse:~/PycharmProjects/finance-buddy$ pip list
Package                Version
---------------------- ------------
annotated-types        0.6.0
anyio                  4.0.0
arrow                  1.3.0
binaryornot            0.4.4
certifi                2023.7.22
chardet                5.2.0
charset-normalizer     3.3.2
click                  8.1.7
colorama               0.4.6
contourpy              1.1.1
cookiecutter           2.4.0
copier                 8.3.0
cycler                 0.12.1
decorator              5.1.1
dunamai                1.19.0
exceptiongroup         1.1.3
flet                   0.12.2
flet-core              0.12.2
flet-runtime           0.12.2
fonttools              4.43.1
funcy                  2.0
h11                    0.14.0
httpcore               0.17.3
httpx                  0.24.1
idna                   3.4
Jinja2                 3.1.2
jinja2-ansible-filters 1.3.2
kiwisolver             1.4.5
markdown-it-py         3.0.0
MarkupSafe             2.1.3
matplotlib             3.8.0
mdurl                  0.1.2
numpy                  1.26.0
oauthlib               3.2.2
packaging              23.2
pandas                 2.1.1
pathspec               0.11.2
Pillow                 10.1.0
pip                    23.3.1
plumbum                1.8.2
prompt-toolkit         3.0.36
pydantic               2.4.2
pydantic_core          2.10.1
Pygments               2.16.1
pyparsing              3.1.1
pypng                  0.20220715.0
python-dateutil        2.8.2
python-slugify         8.0.1
pytz                   2023.3.post1
PyYAML                 6.0.1
pyyaml-include         1.3.1
qrcode                 7.4.2
questionary            2.0.1
repath                 0.9.0
requests               2.31.0
rich                   13.7.0
seaborn                0.13.0
setuptools             65.5.1
six                    1.16.0
sniffio                1.3.0
text-unidecode         1.3
types-python-dateutil  2.8.19.14
typing_extensions      4.8.0
tzdata                 2023.3
urllib3                2.1.0
watchdog               3.0.0
wcwidth                0.2.8
websocket-client       1.6.4
websockets             11.0.3
wheel                  0.38.4
```

**Comment by ndonkoHenri** [comment]
Can you attach the app's code?

**Comment by aurosevic** [comment]
Hey @ndonkoHenri, sure!

Here's a link to a smaller version of the project mentioned above:
https://gist.github.com/aurosevic/8cc6c2713be2fb020fc492781a7c567e

I've identified the problem, but I'm still searching for a solution. The issue arises from line 18 in the provided gist, causing the Tabs structure to break. When ScrollMode is defined on the same page as Tabs, Tabs do not function properly.

This becomes problematic when I want to have Tabs and some other elements simultaneously on a page. Do you know the reason behind this behavior, and is there a solution other than restricting the use of ScrollMode on pages with Tabs?

**Comment by aurosevic** [comment]
@FeodorFitsner @ndonkoHenri Any idea what could be the problem?

**Comment by ndonkoHenri** [comment]
I was able to reproduce the original report on 0.12.2 and re-test on the latest Flet (v0.86.5).

**What's actually going on:** this isn't about `NavigationBar` or navigating away and back. When `page.scroll` is set, the page's content column becomes scrollable, which gives its children *unbounded height*. A `Tabs` control without an explicit height can't be laid out under unbounded constraints, so Flutter's layout fails and the page containing it renders blank/unresponsive — right from launch. That also matches @aurosevic's own finding that the `page.scroll = ft.ScrollMode.ADAPTIVE` line was the trigger.

**Status on latest Flet (v0.86.5):** the same constraint rule applies, but it's easy to handle — give `Tabs` a bounded height, either via `height=...` or `expand=True`. Here's the scenario from this issue, working on the current API, with `page.scroll` set and tabs remaining fully functional after switching navigation destinations back and forth:

```python
import flet as ft


def main(page: ft.Page):
    page.scroll = ft.ScrollMode.ADAPTIVE

    def overview_view():
        return [
            ft.Text("Hello", weight=ft.FontWeight.BOLD, size=30),
            ft.Tabs(
                length=2,
                selected_index=0,
                height=300,  # bounded height — required on a scrollable page
                content=ft.Column(
                    controls=[
                        ft.TabBar(
                            tabs=[
                                ft.Tab(label="Monthly Overview"),
                                ft.Tab(
                                    label="Account Overview", icon=ft.Icons.SETTINGS
                                ),
                            ]
                        ),
                        ft.TabBarView(
                            expand=True,
                            controls=[
                                ft.Container(
                                    alignment=ft.Alignment.CENTER,
                                    content=ft.Text("This is Tab 1"),
                                ),
                                ft.Container(
                                    alignment=ft.Alignment.CENTER,
                                    content=ft.Text("This is Tab 2"),
                                ),
                            ],
                        ),
                    ],
                ),
            ),
        ]

    def groups_view():
        return [
            ft.Column(
                spacing=20,
                controls=[
                    ft.Container(
                        content=ft.Text(value=group),
                        alignment=ft.Alignment.CENTER,
                        height=90,
                        bgcolor=ft.Colors.BLUE_ACCENT_100,
                        border_radius=ft.BorderRadius.all(20),
                    )
                    for group in ("Jan", "Feb")
                ],
            )
        ]

    def change_view(e: ft.Event[ft.NavigationBar] = None):
        label = page.navigation_bar.destinations[
            page.navigation_bar.selected_index
        ].label
        page.controls.clear()
        page.controls.extend(overview_view() if label == "Overview" else groups_view())
        page.update()

    page.navigation_bar = ft.NavigationBar(
        selected_index=0,
        on_change=change_view,
        destinations=[
            ft.NavigationBarDestination(
                icon=ft.Icons.ACCOUNT_BALANCE_ROUNDED, label="Overview"
            ),
            ft.NavigationBarDestination(icon=ft.Icons.GROUPS_ROUNDED, label="Groups"),
        ],
    )
    change_view()


ft.run(main)
```

<img width="710" height="694" alt="Image" src="https://github.com/user-attachments/assets/a02c7daa-75d4-40e3-89d8-bfdf6e4f7873" />

So: setting `page.scroll` alongside `Tabs` is fine on current Flet, as long as the `Tabs` control has a bounded height — no need for the "reset scroll to `None`" workaround anymore. What remains unsatisfying is that an unbounded-height `Tabs` on a scrollable page still fails *silently* (blank page, no error surfaced); will have a look at what we can do. Closing this one — please open a new issue if you hit tab breakage on the latest version even with a bounded height.
